### PR TITLE
Feature: add Gaussian sampler

### PR DESCRIPTION
**Pull request by hyiltiz**
- Add a Gaussian sampler
- Add a convenience function `sample-n` that converts a sampler to a generator

### CODE REVIEW

**Review by pepe — CHANGES_REQUESTED** [review]
Consider adding a test for the new functionality.

**Comment by sogaiu** [comment]
[This line](https://github.com/janet-lang/spork/blob/0b5c8fc0dd21318d0254b1356bc0abad088e74dd/spork/randgen.janet#L45) doesn't look right.

**Comment by sogaiu** [comment]
Not sure what's better, but may be `forever` could be used like this:

```janet
(defn rand-gaussian
  "Get a random sample from the standard Gaussian distribution.
  Optionall specify the mean m and the standard deviation sd. E.g.:
  (rand-gaussian) # => 0.1324...
  (rand-gaussian 5 0.1) # => 5.3397...
  "
  [&opt m sd]
  (default [m sd] [0 1])
  (defn scale [x] (+ m (* sd x)))

  (forever
    (def p (rand-uniform))
    (def q (rand-uniform))

    # We use the Box-Muller transform
    (let [rho (math/sqrt (* -2 (math/log q)))
          theta (* 2 math/pi p)
          _box    (* rho (math/cos theta))
          _muller (* rho (math/sin theta))
          box    (scale _box)
          muller (scale _muller)]

      (yield box)
      (yield muller))))
```

**Comment by sogaiu** [comment]
Regarding the testing comment above, the examples in the docstrings could be broken out as tests.

File: spork/randgen.janet [hunk 1]
```diff
@@ -28,6 +28,44 @@
...
+  "
+  [&opt m sd]
+  (default [m sd] [0 1])
+  (defn scale [x] (+ m (* sd x)))
+
+  (def p (rand-uniform))
+
+  (def q (rand-uniform))
+
+  (math/rng-uniform ))
```
[the remark this answers is not in the document]
Comment (hyiltiz) [reply spork/randgen.janet]: Formatted; thanks for the recommendation!

**Comment by hyiltiz** [comment]
- removed leftover code
- reformatted with `spork/fmt`
- added a test following #165

**Comment by hyiltiz** [comment]
Currently, we have the following problems: 
- PA: evaluating it with same rng multiple times for testing and reproduce; 
- PB: re-use muller values for efficiency; 
- PC: can be called alone and with sample-n

@sogaiu proposed [an alternative implementation](https://gist.github.com/sogaiu/ec2d4387d710d20d2169ef289dcaa52a) using a "closure"-like mechanism.

My original implementation via fibers (using `yield`) couldn't satisfy PC. I think this implementation satisfies them all.

**Comment by sogaiu** [comment]
@hyiltiz and I are comparing the implementation above with his simpler one:

```janet
(defn rand-gaussian [&opt m sd]
  (default m 0)
  (default sd 1)
  (defn scale [x] (+ m (* sd x)))

  (def p (math/rng-uniform (get-rng)))
  (def q (math/rng-uniform (get-rng)))

  # We use the Box-Muller transform
  (let [rho (math/sqrt (* -2 (math/log q)))
        theta (* 2 math/pi p)
        _box (* rho (math/cos theta))
        box (scale _box)]
    box))
```

This latest one is faster and simpler:

```
(set-seed 1)

(timeit-loop [:timeout 1] "simpler" (rand-gaussian-simpler))

(set-seed 1)

(timeit-loop [:timeout 1] "complex" (rand-gaussian-complex))

# simpler 1.000s, 0.485µs/body
# complex 1.000s, 0.8271µs/body
```

**Pull request closed without merging**